### ISSUE TITLE
Slow performance for GET /collections request

### ISSUE THREAD

**Issue by cd-rite** [NUWCDIVNPT/stig-manager#791]
#Query is joining against asset and stig_asset_map table for all requests, even when not required, causing performance issues for very large collections.

**Comment by cd-rite** [comment]
I've found returning the c.settings value directly and substituting missing properties with the defaults in javascript rather than with JSON_MERGE_PATCH in the query also increased performance. (Though not as much as removing the joins to asset, etc).
w db managing:
115 collections
14k assets
timing went from ~1.4sec to ~.5 sec.
(improvement was less dramatic after removing asset joins)


```
      // Loop through each Collection object and merge missing settings before returning data

      collectionResults.forEach(collection => {
        mergeSettings(MyController.defaultSettings, collection.settings);
      });

~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~

//merge existing settings with default values object.
// Function to merge default settings into a given settings object
function mergeSettings(defaults, settings) {
  for (const key in defaults) {
    if (defaults.hasOwnProperty(key)) {
      if (typeof defaults[key] === 'object' && !Array.isArray(defaults[key])) {
        if (!settings[key]) {
          settings[key] = {};
        }
        mergeSettings(defaults[key], settings[key]);
      } else {
        if (!settings.hasOwnProperty(key)) {
          settings[key] = defaults[key];
        }
      }
    }
  }
}
```

**Comment by csmig** [comment]
Thanks for discovering this, deep object merges are way less efficient in MySQL compared to Node! However, I do not recommend bringing in that new code. 

The current approach with the JSON_MERGE was introduced by 6ed01d0319c5e2e83384071a1db95724ccf18c93, which added the `history` property to the Collection settings. In that commit we did not implement a migration to merge the new setting's default into existing Collections. Probably because we thought we'd have to involve the `review_history` table and didn't want to risk the performance hit?

But a JSON_MERGE with the default settings is performed during Collection POST, PUT and PATCH, so `queryCollections()` should be able to select and return `c.settings` as is. After nearly two years, there are probably very few Collections without a complete settings object and any that still exist could be updated by a migration which only affects those outliers.

**Comment by cd-rite** [comment]
resolved by: #1366
